src/legacy/core_plugins/data/public/filter/filter_manager/filter_manager.test.ts

```diff
@@ -0,0 +1,364 @@
+      updateSubscription = filterManager.getUpdates$().subscribe(updateListener);
+      await filterManager.removeAll();
+      expect(updateListener.called).toBeFalsy();
+      expect(filterManager.getFilters()).toHaveLength(0);
+    });
+
+    test('remove on full should clean and fire events', async () => {
+      const f1 = getFilter(FilterStateStore.GLOBAL_STATE, false, false, 'age', 34);
+      const f2 = getFilter(FilterStateStore.APP_STATE, false, false, 'gender', 'FEMALE');
+      await filterManager.setFilters([f1, f2]);
+
+      updateSubscription = filterManager.getUpdates$().subscribe(updateListener);
+      await filterManager.removeAll();
+      expect(updateListener.called).toBeTruthy();
+      expect(filterManager.getFilters()).toHaveLength(0);
+    });
+
+    test('remove non existing filter should do nothing and not fire events', async () => {
+      const f1 = getFilter(FilterStateStore.GLOBAL_STATE, false, false, 'age', 34);
+      const f2 = getFilter(FilterStateStore.APP_STATE, false, false, 'gender', 'FEMALE');
+      const f3 = getFilter(FilterStateStore.APP_STATE, false, false, 'country', 'US');
+      await filterManager.setFilters([f1, f2]);
+      expect(filterManager.getFilters()).toHaveLength(2);
+
+      updateSubscription = filterManager.getUpdates$().subscribe(updateListener);
+      await filterManager.removeFilter(f3);
+      expect(updateListener.called).toBeFalsy();
+      expect(filterManager.getFilters()).toHaveLength(2);
+    });
+
+    test('remove existing filter should remove and fire events', async () => {
+      const f1 = getFilter(FilterStateStore.GLOBAL_STATE, false, false, 'age', 34);
+      const f2 = getFilter(FilterStateStore.APP_STATE, false, false, 'gender', 'FEMALE');
+      const f3 = getFilter(FilterStateStore.APP_STATE, false, false, 'country', 'US');
+      await filterManager.setFilters([f1, f2, f3]);
+      expect(filterManager.getFilters()).toHaveLength(3);
+
+      updateSubscription = filterManager.getUpdates$().subscribe(updateListener);
+      await filterManager.removeFilter(f3);
+      expect(updateListener.called).toBeTruthy();
+      expect(filterManager.getFilters()).toHaveLength(2);
+    });
+  });
+
+  describe('invert', () => {
+    test('invert to disabled', async () => {
+      const f1 = getFilter(FilterStateStore.GLOBAL_STATE, false, false, 'age', 34);
+      filterManager.invertFilter(f1);
+      expect(f1.meta.negate).toBe(true);
+      filterManager.invertFilter(f1);
+      expect(f1.meta.negate).toBe(false);
+    });
+  });
+
+  describe('addFiltersAndChangeTimeFilter', () => {
+    test('should just add filters if there is no time filter in array', async () => {
+      const f1 = getFilter(FilterStateStore.GLOBAL_STATE, false, false, 'age', 34);
+      await filterManager.addFiltersAndChangeTimeFilter([f1]);
+      expect(filterManager.getFilters()).toHaveLength(1);
+    });
+
+    test('TODO: should set timepicker and add filters', () => {});
```

Comment:
TODO needs to be resolved / removed. 

There are a couple others in the PR as well.